bdns/dns.go

```diff
@@ -619,3 +640,48 @@ func logDNSError(
 			underlying)
 	}
 }
+
+type dohExchanger struct {
+	hc http.Client
+}
+
+// Exchange sends a DoH query to the provided DoH server and returns the response.
+func (d *dohExchanger) Exchange(query *dns.Msg, server string) (*dns.Msg, time.Duration, error) {
+	q, err := query.Pack()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// The default Unbound URL template
```

Comment:
Is this the first thing that explicitly binds (heh) us to using Unbound? If, say, Certainly wants to use DoH, but doesn't use Unbound, will this need to become configurable?

Reply:
I don't believe this binds us (boo) to unbound. [RFC 8484 section 4.1.1](https://datatracker.ietf.org/doc/html/rfc8484#section-4.1.1) states that the way to make a DOH query using an HTTP exchange is via the `/dns-query` URI.

Reply:
[PowerDNS](https://dnsdist.org/guides/dns-over-https.html), Unbound, and [bind](https://bind9.readthedocs.io/en/latest/reference.html#interfaces) all set `/dns-query` as the default doh listener endpoint.